sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -407,7 +410,17 @@ private[sql] class ParquetRelation2(
       val filesToTouch =
         if (shouldMergeSchemas) {
           // Also includes summary files, 'cause there might be empty partition directories.
-          (metadataStatuses ++ commonMetadataStatuses ++ dataStatuses).toSeq
+
+          // If skipMergePartFiles config is true, we assume that all part-files are the same for
+          // their schema with summary files, so we ignore them when merging schema.
+          // If the config is false, which is the default setting, we merge all part-files.
+          val needMerged: Seq[FileStatus] =
```

Comment:
However, this feature is still pretty appealing. Can we find out all partition directories containing no summary files and always merge part-files within these directories?